src/Xrefcheck/CLI.hs

```diff
@@ -99,9 +99,16 @@ optionsParser = do
         short 'v' <>
         long "verbose" <>
         help "Report repository scan and verification details."
-    oShowProgressBar <- fmap not . switch $
-        long "no-progress" <>
-        help "Do not display progress bar during verification."
+    oShowProgressBar <- asum
+        [ flag' (Just True) $
+            long "progress" <>
+            help "Display progress bar during verification. \
+                 \This is enabled by default unless `CI` env var is set to true."
+        , flag' (Just False) $
+            long "no-progress" <>
+            help "Do not display progress bar during verification."
+        , pure Nothing
```

Comment:
What will happen if I pass both (maybe multiple times)? I guess it should be documented somewhere.

Reply:
optparse-applicative prompts an error if it sees an extra option, and the same applies for "passed option multiple times" or "pass contradicting options" cases. (the latter case seems to be a corner case of an extra option because once I parsed "--progress", I forget that "--no-progress" was ever expected).

Reply:
Are `--progress` and `--no-progress` considered contradicting in this case?

Reply:
Yes. If I pass `--progress --no-progress`, I get `Invalid option '--no-progress'`.

Reply:
AFAIU more common behavior is to consider the last option, but I think it's not so important at this stage. For example, `stack build --test --no-test --no-test --test` is fine and runs tests, but `stack build --test --no-test --no-test --test --no-test` is also fine and it doesn't run tests.

Reply:
I will do this because it may be convenient, but some later, because once I rebased on top of lts-15 I realized that I cannot build without exceeding my hard drive memory.